src/frontend/state/InjectedProviderHandler.ts

```diff
@@ -0,0 +1,159 @@
+import { makeAutoObservable } from 'mobx'
+
+class InjectedProviderHandler {
+  ethereumListenersBound = false
+
+  async handleRequest(
+    event: Electron.IpcRendererEvent,
+    id: number,
+    args: Record<string, unknown>
+  ) {
+    try {
+      await this.waitForConnection()
+      /* eslint-disable-next-line */
+      //@ts-ignore
+      const value = await window.ethereum?.request(args)
+      window.api.returnExtensionRequest(id, value)
+    } catch (err) {
+      console.error(`error during request: ${err}`)
+      window.api.errorExtensionRequest(id, err)
```

Comment:
Is this going to sentry as well?

Reply:
No. This returns the error to whoever called this provider whether that be a proxy server call or a window.ethereum call somewhere in the app